sam/record.go

```diff
@@ -490,3 +516,32 @@ func (ns Seq) Expand() []byte {
 
 	return s
 }
+
+// Base returns the pos'th base of the sequence.  "pos" must be smaller than
+// seq.Length.
+func (ns Seq) Base(pos int) SeqBase {
```

Comment:
I would prefer this be named `At` and return a `byte` which is actually the encoded letter, so just what your `BaseChar` does below.

The documentation here should state that the call will panic if `pos < 0 || ns.Len() <= pos`, and do this explicit check and panic with an informative message.